shared_model/backend/protobuf/impl/proto_query_response_factory.cpp

```diff
@@ -0,0 +1,230 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "backend/protobuf/proto_query_response_factory.hpp"
+#include "backend/protobuf/permissions.hpp"
+#include "backend/protobuf/query_responses/proto_block_query_response.hpp"
+#include "backend/protobuf/query_responses/proto_query_response.hpp"
+
+std::unique_ptr<shared_model::interface::QueryResponse>
+shared_model::proto::ProtoQueryResponseFactory::createAccountAssetResponse(
+    std::vector<std::shared_ptr<shared_model::interface::AccountAsset>> assets,
+    const crypto::Hash &query_hash) {
+  iroha::protocol::QueryResponse protocol_query_response;
+  protocol_query_response.set_query_hash(query_hash.hex());
+
+  iroha::protocol::AccountAssetResponse *protocol_specific_response =
+      protocol_query_response.mutable_account_assets_response();
+  for (const auto &asset : assets) {
+    protocol_specific_response->add_account_assets()->CopyFrom(
```

Comment:
No need to `CopyFrom`, `*protocol_specific_response->add_account_assets() = getTransport()` calls copy constructor. Same for other methods.